Src/DFC.App.MatchSkills.Services.ServiceTaxonomy/ServiceTaxonomyRepository.cs

```diff
@@ -1,8 +1,70 @@
-﻿using DFC.App.MatchSkills.Application.ServiceTaxonomy;
+﻿using System;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using AutoMapper;
+using DFC.App.MatchSkills.Application.ServiceTaxonomy;
+using DFC.App.MatchSkills.Services.ServiceTaxonomy.Models;
+using DFC.Personalisation.Common.Net.RestClient;
+using DFC.Personalisation.Domain.Models;
+
 
 namespace DFC.App.MatchSkills.Services.ServiceTaxonomy
 {
     public class ServiceTaxonomyRepository : IServiceTaxonomyReader, IServiceTaxonomySearcher
     {
+        private static  HttpClient _httpClient=default;
+        public ServiceTaxonomyRepository(HttpClient httpClient)
+        {
+            _httpClient = httpClient;
+        }
+        static async Task<TList> GetJsonList<TList>(string apiPath, string ocpApimSubscriptionKey) where TList : class
+        { 
+            if (string.IsNullOrWhiteSpace((apiPath)))
+                throw new ArgumentNullException(nameof(apiPath), "ApiPath must be specified.");
+
+            if (string.IsNullOrWhiteSpace((ocpApimSubscriptionKey)))
+                throw new ArgumentNullException(nameof(ocpApimSubscriptionKey),
+                    "Ocp-Apim-Subscription-Key must be specified.");
+
+            var restClient = new RestClient(_httpClient);
+            return await restClient.Get<TList>(apiPath,ocpApimSubscriptionKey);
+            
+        }
+       
+        public async Task<Skill[]> GetAllSkills<TSkills>(string apiPath, string ocpApimSubscriptionKey)
+        {
+            var result = await GetJsonList<STSkills>(apiPath, ocpApimSubscriptionKey); 
+            return Mapping.Mapper.Map<Skill[]>(result.Skills);
+        }
+
+        
+        public async Task<Occupation[]> GetAllOccupations<TOccupations>(string apiPath, string ocpApimSubscriptionKey) 
```

Comment:
I'd suggest renaming GetAllOccupations and GetAllSkills to just GetAll. Semantically they are performing identical operations just on different data sets.
Same with Search.